File: FPG.Core/Factories/SymbolInsertionAlgorithmFactory.cs
using System;

using FPG.Algorithms;
using FPG.Interfaces;

namespace FPG.Factories
{
    public static class SymbolInsertionAlgorithmFactory
    {
        public static ISymbolInsertionAlgorithm Create(String name)
        {
            switch (name.ToLower())
            {
                case "npicked":
                    return new NPickedSymbolInsertionAlgorithm();
                default:
                    throw new ArgumentException("Unknown algorithm name");
            };
        }
    }
}

File: lib/core/factories/SymbolInsertionAlgorithmFactory.dart
import 'package:fpg/core/algorithms/symbolinsertion/NPickedSymbolInsertionAlgorithm.dart';
import 'package:fpg/core/interfaces/ISymbolInsertionAlgorithm.dart';

class SymbolInsertionAlgorithmFactory {
  static ISymbolInsertionAlgorithm create(String name) {
    switch (name.toLowerCase()) {
      case "npicked":
        return NPickedSymbolInsertionAlgorithm();
      default:
        throw FormatException("Unknown algorithm name");
    }
  }
}
